src/test/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/letter/InternalEndAppealTemplateTest.java

```diff
@@ -0,0 +1,106 @@
+    private String homeOfficeReferenceNumber = "123654";
+    private String appealReferenceNumber = "HU/11111/2022";
+    private String appealEndDate = "2023-07-01";
+    private String judgeName = "Mr.Test";
+    private final String templateName = "INTERNAL_END_APPEAL_NOTICE_TEMPLATE.docx";
+    private final String logo = "[userImage:hmcts.png]";
+    private InternalEndAppealTemplate internalEndAppealTemplate;
+    private Map<String, Object> fieldValuesMap;
+
+    @BeforeEach
+    public void setUp() {
+        internalEndAppealTemplate =
+                new InternalEndAppealTemplate(templateName, customerServicesProvider);
+    }
+
+    @Test
+    void should_return_template_name() {
+        assertEquals(templateName, internalEndAppealTemplate.getName());
+    }
+
+    void dataSetup() {
+        when(caseDetails.getCaseData()).thenReturn(asylumCase);
+        when(customerServicesProvider.getInternalCustomerServicesTelephone(asylumCase)).thenReturn(telephoneNumber);
+        when(customerServicesProvider.getInternalCustomerServicesEmail(asylumCase)).thenReturn(adaEmail);
+        when(asylumCase.read(APPEAL_REFERENCE_NUMBER, String.class)).thenReturn(Optional.of(appealReferenceNumber));
+        when(asylumCase.read(APPELLANT_GIVEN_NAMES, String.class)).thenReturn(Optional.of(appellantGivenNames));
+        when(asylumCase.read(APPELLANT_FAMILY_NAME, String.class)).thenReturn(Optional.of(appellantFamilyName));
+        when(asylumCase.read(HOME_OFFICE_REFERENCE_NUMBER, String.class)).thenReturn(Optional.of(homeOfficeReferenceNumber));
+        when(asylumCase.read(END_APPEAL_DATE, String.class)).thenReturn(Optional.of(appealEndDate));
+        when(asylumCase.read(END_APPEAL_APPROVER_NAME, String.class)).thenReturn(Optional.of(judgeName));
+        when(asylumCase.read(IS_ACCELERATED_DETAINED_APPEAL, YesOrNo.class)).thenReturn(Optional.of(YesOrNo.YES));
+    }
+
+    @Test
+    void should_populate_template_for_ada() {
+        dataSetup();
+        fieldValuesMap = internalEndAppealTemplate.mapFieldValues(caseDetails);
+        assertEquals(logo, fieldValuesMap.get("hmcts"));
+        assertEquals(appealReferenceNumber, fieldValuesMap.get("appealReferenceNumber"));
+        assertEquals(appellantGivenNames, fieldValuesMap.get("appellantGivenNames"));
+        assertEquals(appellantFamilyName, fieldValuesMap.get("appellantFamilyName"));
+        assertEquals(homeOfficeReferenceNumber, fieldValuesMap.get("homeOfficeReferenceNumber"));
+        assertEquals(telephoneNumber, fieldValuesMap.get("customerServicesTelephone"));
+        assertEquals(adaEmail, fieldValuesMap.get("customerServicesEmail"));
+        assertEquals(adaFormName, fieldValuesMap.get("formName"));
+        assertEquals(LocalDate.now().format(DateTimeFormatter.ofPattern("d MMM yyyy")), fieldValuesMap.get("dateLetterSent"));
+        assertEquals(LocalDate.now().format(DateTimeFormatter.ofPattern("d MMM yyyy")), fieldValuesMap.get("dateLetterSent"));
+    }
+
+    @Test
```

Comment:
We can use paramterized testing here if you want to switch between ADA and non-ADA.

Reply:
done